packages/cypress/src/generators/migrate-to-cypress-ten/conversion.util.ts

```diff
@@ -0,0 +1,347 @@
+  );
+  const newIntegrationFolder = joinPathFragments(
+    projectConfig.root,
+    integrationFolder
+  );
+
+  let newSupportFile: string;
+  let oldSupportFile: string;
+  let oldImportLeafPath: string;
+  let newImportLeafPath: string;
+  let shouldUpdateSupportFileImports = false;
+  // supportFile can be falsey or a string path to the file
+  if (cypressConfigJson.supportFile) {
+    // we need to check the test files to see if
+    // support file import is used and then update it if so
+    shouldUpdateSupportFileImports = true;
+    oldSupportFile = joinPathFragments(
+      projectConfig.root,
+      cypressConfigJson.supportFile
+    );
+
+    newSupportFile = joinPathFragments(projectConfig.root, supportFile);
+    // oldSupportFile might have already been updated
+    // to the new default location so must be guarded for
+    if (oldSupportFile !== newSupportFile && tree.exists(oldSupportFile)) {
+      tree.rename(oldSupportFile, newSupportFile);
+    }
+  } else {
+    shouldUpdateSupportFileImports = false;
+    newSupportFile = supportFile;
+    // rename the default support file even if not in use to keep the system in sync with cypress v10
+    const defaultSupportFile = joinPathFragments(
+      projectConfig.sourceRoot,
+      'support',
+      'index.ts'
+    );
+
+    if (tree.exists(defaultSupportFile)) {
+      const newSupportDefaultPath = joinPathFragments(
+        projectConfig.sourceRoot,
+        'support',
+        'e2e.ts'
+      );
+      if (
+        defaultSupportFile !== newSupportDefaultPath &&
+        tree.exists(defaultSupportFile)
+      ) {
+        tree.rename(defaultSupportFile, newSupportDefaultPath);
+      }
+    }
+  }
+
+  if (shouldUpdateSupportFileImports) {
+    const newImportPaths = createSupportFileImport(
+      oldSupportFile,
+      newSupportFile,
+      projectConfig.sourceRoot
+    );
+    oldImportLeafPath = newImportPaths.oldImportPathLeaf;
+    newImportLeafPath = newImportPaths.newImportPathLeaf;
+  }
+
+  // tree.rename doesn't work on directories must update each file within
+  // the directory to the new directory
+  visitNotIgnoredFiles(tree, projectConfig.sourceRoot, (path) => {
+    if (!path.includes(oldIntegrationFolder)) {
+      return;
+    }
+    const fileName = basename(path);
+    let newPath = path.replace(oldIntegrationFolder, newIntegrationFolder);
+
+    if (fileName.includes('.spec.')) {
+      newPath = newPath.replace('.spec.', '.cy.');
+    }
+    if (newPath !== path && tree.exists(path)) {
+      tree.rename(path, newPath);
+    }
+    // if they weren't using the supportFile then there is no need to update the imports.
+    if (
+      shouldUpdateSupportFileImports &&
+      validFilesEndingsToUpdate.some((e) => path.endsWith(e))
+    ) {
+      updateImports(tree, newPath, oldImportLeafPath, newImportLeafPath);
+    }
+  });
+
+  if (tree.children(oldIntegrationFolder).length === 0) {
+    tree.delete(oldIntegrationFolder);
+  }
+}
```

Comment:
@FrozenPandaz we should add a `renameFolderInTree` util

Reply:
in this case, it makes sense to use the `visitNotIgnoredFiles` since I need to do more than just move the files, but always down for more helper utilities.